### PR TITLE
feat: upgrade Rust from 1.75 to 1.76

### LINKED ISSUE #583: Problem: use the latest stable release of the rust toolchain
Upgrade the rust toolchain from 1.75 to 1.76. Modify rust-toolchain.toml and dockerfile.

### PR DESCRIPTION
**Pull request by jhult**
Closes #583

### CODE REVIEW

File: src/block/parser.rs [hunk 1]
```diff
@@ -123,7 +123,7 @@ impl BlockParser {
                 orphaned_paths: vec![].into_iter(),
             })
         } else {
-            Ok(Self::empty(blocks_dir, &vec![]))
+            Ok(Self::empty(blocks_dir, &[]))
```
Comment (robinbb) [inline src/block/parser.rs:126]: I don't know enough to be able to tell if this change is related to the Rust minor version change. Is it?

File: src/block/parser.rs [hunk 2]
```diff
@@ -237,13 +237,13 @@ impl BlockParser {
...
 
-    fn empty(blocks_dir: &Path, paths: &Vec<PathBuf>) -> Self {
+    fn empty(blocks_dir: &Path, paths: &[PathBuf]) -> Self {
         Self {
             num_canonical: 0,
             total_num_blocks: paths.len() as u32,
             blocks_dir: blocks_dir.to_path_buf(),
             canonical_paths: vec![].into_iter(),
-            recent_paths: paths.clone().into_iter(),
+            recent_paths: paths.to_owned().into_iter(),
```
Comment (robinbb) [inline src/block/parser.rs:246]: Is this change related to the Rust minor version change?

File: src/block/parser.rs (hunk 1 above)
Comment (trevorbernard) [reply src/block/parser.rs:126]: There are new lints when upgrading to the latest stable

File: src/block/parser.rs (hunk 2 above)
Comment (trevorbernard) [reply src/block/parser.rs:246]: See previous comment

**Comment by trevorbernard** [comment]
Looks like one lint was missed. Moving to draft until it's resolved.

```rust

error: unnecessary use of `to_owned`
--
  | --> src/block/parser.rs:246:27
  | \|
  | 246 \|             recent_paths: paths.to_owned().into_iter(),
  | \|                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `paths.iter().cloned()`
  | \|
  | = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_to_owned
  | = note: `-D clippy::unnecessary-to-owned` implied by `-D warnings`
  | = help: to override `-D warnings` add `#[allow(clippy::unnecessary_to_owned)]`


```

**Comment by jhult** [comment]
Ran into [problems with Clippy](https://github.com/rust-lang/rust-clippy/issues/8148). Ended up using `Vec::from()` syntax (since `to_vec()` triggered Clippy)

**Review by trevorbernard — APPROVED** [review]
:shipit:
